src/StoreKeeper/WooCommerce/B2C/Helpers/Seo/YoastSeo.php

```diff
@@ -0,0 +1,200 @@
+<?php
+
+namespace StoreKeeper\WooCommerce\B2C\Helpers\Seo;
+
+use StoreKeeper\WooCommerce\B2C\Exceptions\WordpressException;
+use StoreKeeper\WooCommerce\B2C\Frontend\Handlers\Seo;
+use StoreKeeper\WooCommerce\B2C\Options\StoreKeeperOptions;
+use StoreKeeper\WooCommerce\B2C\Tools\WordpressExceptionThrower;
+
+/**
+ * Tested up to Yoast SEO version 17.8.
+ */
+class YoastSeo
+{
+    public const YOAST_SEO_OPTION_KEY = 'wpseo_taxonomy_meta';
+
+    /**
+     * @throws WordpressException
+     */
+    public static function getPostTitle(int $postId): string
+    {
+        $post = WordpressExceptionThrower::throwExceptionOnWpError(
+            get_post($postId)
+        );
+        $yoastSeoTitle = '';
+        if (Seo::isYoastActive() && !is_null($post)) {
```

Comment:
```suggestion
        if (self::isActive() && !is_null($post)) {
```
It would make more sense to have the function here, now you are calling some frontend class in a helper

Reply:
Moved checking of yoast to `PluginStatus`.